src/thor/astar_bss.cc

```diff
@@ -41,11 +42,17 @@ AStarBSSAlgorithm::~AStarBSSAlgorithm() {
 
 // Clear the temporary information generated during path construction.
 void AStarBSSAlgorithm::Clear() {
+  // Reduce edge labels capacity if it's more than limit
+  if (edgelabels_.size() > max_reserved_labels_count_) {
+    edgelabels_.resize(max_reserved_labels_count_);
+    edgelabels_.shrink_to_fit();
+  }
   // Clear the edge labels and destination list. Reset the adjacency list
   // and clear edge status.
   edgelabels_.clear();
   destinations_.clear();
-  adjacencylist_.reset();
+  if (adjacencylist_)
```

Comment:
i didnt checking into it at all, but since we always want to reuse it via `clear` can we also make it nonpointer?

Reply:
originally `DoubleBucketQueue` doesn't have a default constructor (because it doesn't make sense). that is the reason why I use a pointer. But from the other hand I agree with you @kevinkreiser , code looks more bulky with `unique_ptr`; will try to recollect it

Reply:
ok. fixed it by introducing (actually, re-introducing) default constructor for DoubleBucketQueue